tests/e2e/lib/diskutils

```diff
@@ -15,48 +15,108 @@
+      #}
+
+      # Sync the system to flush any cached I/O operations to the disk
+      # This ensures that all file system operations are complete before modifying partitions
+      echo "Calling sync before partitioning /dev/${disk}..."
+      sync
+
+      # Proceed with partitioning using fdisk
+      # fdisk is used here in non-interactive mode to create a new primary partition
+      echo "Creating new partition on /dev/${disk}..."
+      new_part="$( (echo n; echo p; echo ; echo ; echo ; echo w) | fdisk "/dev/${disk}")" || {
+        echo "Failed to create partition on /dev/${disk}, skipping."
+        continue
+      }
+
+      # Extract the partition number from the fdisk output
+      part_id="$(echo "${new_part}" | grep -Po "new partition \K([0-9])")"
+      DISK="${disk}"
+
+      # Check if the disk is an NVMe device (which has partition naming like /dev/nvme0n1p1)
+      # Append the partition ID correctly for NVMe disks
+      if [[ $(echo "${DISK}" | grep -c nvme) -eq 1 ]]; then
+        PART_ID="p${part_id}"
+        info_message "select_disk_to_partition, found /dev/${DISK}${PART_ID}"
+        info_message "=============================="
+        break  # Stop further processing as we have found and partitioned the desired disk
+      else
+        PART_ID="${part_id}"
+        info_message "select_disk_to_partition, found /dev/${DISK}${PART_ID}"
+        info_message "=============================="
+        break  # Stop further processing after partitioning
       fi
-      if [ -e "${SOC_DISTRO_FILE}" ]; then
-        if grep -qi "${QC_SOC}" "${SOC_DISTRO_FILE}"; then
-           if [[ "${disk}" == "${QC_SOC_DISK}" ]]; then
-             DISK=${disk}
-             part=$(echo "${disk_table}" | grep "G" | grep "${QC_SOC_DISK}[0-9]." | grep -v "/" | cut -d" " -f1 )
-             PART_ID=${part:3}
-             break;
-           fi
+    fi
+
+    # Additional logic for SOC-specific disk selection (if necessary)
+    if [ -e "${SOC_DISTRO_FILE}" ]; then
+      if grep -qi "${QC_SOC}" "${SOC_DISTRO_FILE}"; then
+        if [[ "${disk}" == "${QC_SOC_DISK}" ]]; then
+          DISK=${disk}
+
+          # Get the partition ID for the SOC-specific disk (partition in GB)
+          part=$(echo "${disk_table}" | grep "G" | grep "${QC_SOC_DISK}[0-9]." | grep -v "/" | cut -d" " -f1)
+          PART_ID="${part:3}"  # Extract partition ID from the result
+          break  # Stop further processing after selecting the SOC-specific disk
         fi
       fi
+    fi
   done
-  parted /dev/${DISK} rescan
+
+  # Once a disk is found and partitioned, we need to notify the kernel about the changes
+  if [[ -n "${DISK}" ]]; then
+    echo "Rescanning partition table on /dev/${DISK}..."
+
+    # Run partprobe to inform the kernel about partition changes
+    # This avoids the need for a reboot, ensuring that partition changes are recognized immediately
+    partprobe /dev/${DISK} || {
```

Comment:
This one should not be called for SoC 
since in the SoC use case, there is not format usage.
just finding disc and mkfs

Reply:
I do not see any change here

Reply:
See the next patch. Its in the flow

Reply:
i do not think
see this 
https://pastebin.com/KJxBak0k
Do you want to partprobe?

Reply:
> i do not think, see this log https://pastebin.com/KJxBak0k Do you want to partprobe?
> 
> this one? [b193491](https://github.com/containers/qm/commit/b193491f7c3882ab0f347bcd6d3aeab2c631494d)
> 
No: https://github.com/containers/qm/pull/538/commits/b193491f7c3882ab0f347bcd6d3aeab2c631494d

Reply:
OK, added here the missing part